test/C/src/federated/HelloWorldFederatedCPPNoCMake.lf

```diff
@@ -0,0 +1,31 @@
+/**
+ * This test is a variant of HelloWorld.lf that checks the compatibility
+ * of the federated C runtime with g++ and no CMake. This test should not pass if it does not compile.
+ */
+target C {
+    tracing: true,
+    compiler: "g++",
+    flags: "-Wno-write-strings", // Suppress a ridiculous number of warnings.
```

Comment:
Why not address the warnings?

Reply:
These warnings are due to a behavior that is undefined under `C++` but perfectly legal in `C`. There is a discussion about this [here](https://stackoverflow.com/questions/20944784/why-is-conversion-from-string-constant-to-char-valid-in-c-but-invalid-in-c). This warning can go away if we explicitly cast string assignments (e.g., `="foo"`) to `char*` (e.g., `=(char*)"foo"`). However, I don't think we should care too much about C++ in the C target though, although we can add these casts and be done with it. Nonetheless, I think this change might merit its own pull request perhaps.

Reply:
OK, I understand. Let's not address this as part of PR then.